src/box/txn_limbo.h

```diff
@@ -189,14 +189,26 @@ struct txn_limbo {
 	 * synchro command (promote/demote/...) fails.
 	 */
 	int64_t svp_confirmed_lsn;
-	/**
-	 * Whether the limbo is frozen. This mode prevents CONFIRMs and
-	 * ROLLBACKs being written by this instance. This mode is turned on when
-	 * quorum is lost if this instance is the current RAFT leader and
-	 * fencing is enabled. Instance leaves this mode when it becomes leader
-	 * again or PROMOTE/DEMOTE arrives from some remote instance.
-	 */
-	bool frozen;
+	union {
+		/**
+		 * Whether the limbo is frozen. This mode prevents CONFIRMs and
+		 * ROLLBACKs being written by this instance. This, in turn,
+		 * helps to prevent split-brain situations, when a node
+		 * finalizes some transaction before knowing that the
+		 * transaction was already finalized by someone else.
+		 */
+		uint8_t is_frozen_reasons;
+		struct {
+			/*
+			 * This mode is turned on when quorum is lost if this
+			 * instance is the current RAFT leader and fencing is
+			 * enabled. Instance leaves this mode when it becomes
+			 * leader again or PROMOTE/DEMOTE arrives from some
+			 * remote instance.
+			 */
+			bool is_frozen_due_to_fencing : 1;
```

Comment:
Core team will smack our asses for bit fields 🌚🌚🌚🌚🌚🌚 (fine for me).

Reply:
Let it be our little secret.